### ISSUE TITLE
latex: the captions of code blocks inside tables are ignored

### ISSUE THREAD

**Issue by jfbu** [sphinx-doc/sphinx#2706]
Test:

``` rst
.. list-table::


   * - .. code-block:: python
          :caption: XXX

          1+1

     - .. code-block:: python
          :caption: YYY

          1+1
          2+2
          3+3

.. cssclass:: longtable

+-----------------------+-----------------------+
| **literalblock**      |                       |
+-----------------------+-----------------------+
|.. code-block:: python | .. code-block:: python|
|   :caption: WWW       |    :caption: ZZZ      |
|                       |                       |
|   1+1                 |    1+1                |
|                       |    2+2                |
+-----------------------+-----------------------+

.. code-block:: python
   :caption: UUU

   1+1
```

produces (tested both on stable and master branches) via `make latexpdf`:

![capture d ecran 2016-06-21 a 09 34 56](https://cloud.githubusercontent.com/assets/2589111/16221575/da443850-3793-11e6-8190-e3db10f5175c.png)

If using PR #2704, this becomes

![capture d ecran 2016-06-21 a 09 36 01](https://cloud.githubusercontent.com/assets/2589111/16221592/f5b54a8e-3793-11e6-9289-1d95b2e5a109.png)

The hyperlink targets work as one can test. Things are not perfect, for example `:linenos:` works but the line numbers by necessity appear inside adjacent cell.

**Comment by jfbu** [comment]
arguably, `\abovecaptionskip` should be reduced when caption is inside a table cell and not at top level of document layout.

**Comment by jfbu** [comment]
Fixed by f1cc694 (PR #2704). The skip above caption has been reduced to `\smallskipamount` when inside table cell.
